torchvision/datasets/utils.py

```diff
@@ -42,14 +42,33 @@ def check_integrity(fpath: str, md5: Optional[str] = None) -> bool:
     return check_md5(fpath, md5)
 
 
-def download_url(url: str, root: str, filename: Optional[str] = None, md5: Optional[str] = None) -> None:
+def _get_redirect_url(url: str, hop_idx: int = 0, max_hops: int = 10) -> str:
+    import requests
+
+    if hop_idx >= max_hops:
+        raise RecursionError(f'Too many redirects: {hop_idx} (max. allowed {max_hops})')
+
+    response = requests.get(url)
+    if response.url != url and response.url is not None:
+        redirect_url = response.url
+        print(f'Redirecting to {redirect_url}')
+
+        return _get_redirect_url(redirect_url, hop_idx + 1)
```

Comment:
Nit: it might be good to avoid a recursive implementation here in favour of a loop.